models/openacademy_session.py

```diff
@@ -1,12 +1,49 @@
-from odoo import models, fields
+from odoo import models, fields, api
+from odoo.exceptions import ValidationError
 
 class Session(models.Model):
     _name = "openacademy.session"
     _description = "This is the description of Session model."
 
     name = fields.Char()
-    start_date = fields.Date()
+    start_date = fields.Date(default=fields.Date.today)
     duration = fields.Integer()
     number_seats = fields.Integer("Number of seats")
-    instructor_id = fields.Many2one("res.partner", ondelete="restrict")
+    instructor_id = fields.Many2one("res.partner", domain="['|', ('instructor', '=', 'True'), ('category_id.name', '=', 'Teacher / Level 1'), ('category_id.name', '=', 'Teacher / Level 2')]", ondelete="restrict")
     course_id = fields.Many2one("openacademy.course", ondelete="restrict", required=True)
+    attendee_ids = fields.Many2many("res.partner", string="Attendees")
+    taken_seats = fields.Integer()
+    perc_taken_seats = fields.Float(digits=(3, 2), compute="_compute_perc_taken_seats", store=True)
+    active = fields.Boolean(default=True)
+
+    @api.depends("taken_seats", "number_seats")
+    def _compute_perc_taken_seats(self):
+        for session in self:
+            if session.number_seats != 0:
+                session.perc_taken_seats = 100*(session.taken_seats/session.number_seats)
+            else:
+                session.perc_taken_seats = 0
+
+    @api.onchange("number_seats", "taken_seats")
+    def _onchange_seats(self):
+        for session in self:
```

Comment:
I think this can be skipped and self used directly, since for onchange functions the change would be applied to only one record at a time unlike computed fields.

Reply:
Okay, got it.